prow/bugzilla/client_test.go

```diff
@@ -1004,74 +1004,78 @@ func errorChecker(err error, t *testing.T) {
 	}
 }
 func TestGetAllClones(t *testing.T) {
-	fake := &Fake{}
-	fake.Bugs = map[int]Bug{}
-	fake.BugComments = map[int][]Comment{}
-	bug1Create := &BugCreate{
-		Summary: "Dummy bug to test getAllClones",
-	}
-	bug1ID, err := fake.CreateBug(bug1Create)
-	if err != nil {
-		t.Fatalf("Error while creating bug in Fake!\n")
-	}
-	bug1, err := fake.GetBug(bug1ID) //Original bug
-	errorChecker(err, t)
-	bug2ID, err := fake.CloneBug(bug1) //2nd level bug
-	errorChecker(err, t)
-	bug2, err := fake.GetBug(bug2ID)
-	errorChecker(err, t)
-	bug3ID, err := fake.CloneBug(bug2) //3rd level bug
-	errorChecker(err, t)
-	bug4ID, err := fake.CloneBug(bug1) //Sibling of 2nd level
-	errorChecker(err, t)
-	bug1, err = fake.GetBug(bug1ID)
-	errorChecker(err, t)
-	bug2, err = fake.GetBug(bug2ID)
-	errorChecker(err, t)
-	bug3, err := fake.GetBug(bug3ID)
-	errorChecker(err, t)
-	bug4, err := fake.GetBug(bug4ID)
-	errorChecker(err, t)
+
 	testcases := []struct {
 		name           string
-		bug            *Bug
-		expectedClones sets.Int
+		bugs           []Bug
+		expectedClones map[int]sets.Int
 	}{
 		{
 			"Clones including multiple children",
-			bug1,
-			sets.NewInt(bug2ID, bug3ID, bug4ID),
-		},
-		{
-			"Clones should include parent as well as child",
-			bug2,
-			sets.NewInt(bug1ID, bug3ID, bug4ID),
-		},
-		{
-			"Clones includes parent and grandparent",
-			bug3,
-			sets.NewInt(bug1ID, bug2ID, bug4ID),
-		},
-		{
-			"Clones when not directly related",
-			bug4,
-			sets.NewInt(bug1ID, bug2ID, bug3ID),
+			[]Bug{
+				{
+					Summary:   "Dummy bug to test getAllClones",
+					ID:        1,
+					DependsOn: []int{},
```

Comment:
You can just leave empty fields out. You can also leave the `Summary` out for everything that should match, empty string will match empty string. You can then probably just define the bug in a single line